geoutils/geovector.py

```diff
@@ -320,3 +324,189 @@ def query(self: VectorType, expression: str, inplace: bool = False) -> VectorTyp
         new_vector = self.__new__(type(self))
         new_vector.__init__(self.ds.query(expression))
         return new_vector  # type: ignore
+
+    def buffer_without_overlap(self, buffer_size: int | float, plot: bool = False) -> np.ndarray:
+        """
+        Returns a Vector object containing self's geometries extended by a buffer, without overlapping each other.
+
+        The algorithm is based upon this tutorial: https://statnmap.com/2020-07-31-buffer-area-for-nearest-neighbour/.
+        The buffered polygons are created using Voronoi polygons in order to delineate the "area of influence" \
+of each geometry.
+        The buffer is slightly inaccurate where two geometries touch, due to the nature of the Voronoi polygons,\
+hence one geometry "steps" slightly on the neighbor buffer in some cases.
```

Comment:
No they won't overlap, but the buffer does not start exactly at the edge between the two contours (because the Voronoi polygons are based on the vertices, so they always start halfway between two vertices. I'll try to show a zoom.

Reply:
This is a zoom from the example above. You can see that the buffer edge (in gray) is not directly in the continuity of the outlines edges (in black)
![Figure_2](https://user-images.githubusercontent.com/3285905/127498830-78f6e684-c32c-41ea-b692-8ee89b67160f.png)

Reply:
I see! Thanks for the explanation.